### PR TITLE
SONAR-7307 retrieve only one row when polling queue table

### PR DESCRIPTION
**Pull request by sns-seb**

### CODE REVIEW

File: sonar-db/src/main/java/org/sonar/db/ce/CeQueueDao.java [hunk 1]
```diff
@@ -33,6 +33,8 @@
 
 public class CeQueueDao implements Dao {
 
+  private static final RowBounds ONE_ROW_LIMIT = new RowBounds(0, 1);
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/ce/CeQueueDao.java]: MAJOR Remove this unused "ONE_ROW_LIMIT" private field. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1068)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: sonar-db/src/main/java/org/sonar/db/ce/CeQueueDao.java [hunk 2]
```diff
@@ -107,7 +109,7 @@ public int countAll(DbSession dbSession) {
   }
 
   public Optional<CeQueueDto> peek(DbSession session) {
-    List<String> taskUuids = mapper(session).selectEligibleForPeek();
+    List<String> taskUuids = mapper(session).selectEligibleForPeek(ONE_ROW_LIMIT);
```
Comment (simonbrandhof) [inline sonar-db/src/main/java/org/sonar/db/ce/CeQueueDao.java]: Using MyBatis RowBounds is not the perfect solution as efficiency depends on driver implementation. In a perfect world the SELECT request should define the LIMIT and OFFSET, but it's not trivial as syntax differs according to db vendor.
Still RowBounds can be considered as good enough here.

**Comment by simonbrandhof** [comment]
LGTM

**Pull request merged**